modules/juce_gui_basics/native/juce_mac_Windowing.mm

```diff
@@ -488,6 +488,7 @@ static void displayReconfigurationCallBack (CGDirectDisplayID, CGDisplayChangeSu
     NSSize dpi = [[[s deviceDescription] objectForKey: NSDeviceResolution] sizeValue];
     d.dpi = (dpi.width + dpi.height) / 2.0;
 
+    d.rawDisplay = s;

```

Comment:
You're not retaining the `NSScreen` object. What if it gets deallocated and you still have a pointer to it?

Reply:
I assume that when a display gets disconnected, JUCE reacts and maintains its array of displays and as such there won't be a pointer to it after that point, probably due to a notification that the screen was disconnected where probably the object wasn't released yet.

Reply:
Makes sense. Worth testing probably

Reply:
As @yairchu wrote. Before committing this I've followed the code flow and juce `Displays::Display` flow.
https://github.com/soundradix/JUCE/blob/421431426047ab21a7f83e685ab29a36d4d2f087/modules/juce_gui_basics/desktop/juce_Displays.cpp#L168

So it might not be thread-safe but the display object life-cycle manages the NSScreen pointer.